app/controllers/thredded/private_topics_controller.rb

```diff
@@ -66,6 +66,12 @@ def update
       end
     end
 
+    def mark_all_read
+      MarkAllRead.run(Thredded::PrivateTopic, thredded_current_user) if signed_in?
```

Comment:
I wasn't sure if we'd do this on public topics so got all "generic". I'm fine with privates only.